oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java

```diff
@@ -84,8 +85,15 @@ public boolean isExists(Model model) throws StorageException {
                 Optional<Index> index = esClient.getIndex(tableName);
                 Mappings historyMapping = index.map(Index::getMappings).orElseGet(Mappings::new);
                 structures.putStructure(tableName, historyMapping, index.map(Index::getSettings).orElseGet(HashMap::new));
-                exist = structures.containsMapping(tableName, createMapping(model))
-                    && structures.compareIndexSetting(tableName, createSetting(model));
+                boolean containsMapping = structures.containsMapping(tableName, createMapping(model));
+                boolean compareIndexSetting = structures.compareIndexSetting(tableName, createSetting(model));
+                // "no-init mode" no needs to check index settings for updating,
+                // to avoid conflicts between "init mode and no-init mode" index settings configurations
+                if (RunningMode.isNoInitMode()) {
+                    exist = containsMapping;
+                } else {
+                    exist = containsMapping && compareIndexSetting;
+                }
```

Comment:
```suggestion
                boolean containsMapping = structures.containsMapping(tableName, createMapping(model));
                // Do not check index settings in the "no-init mode",
                // because the no-init mode OAP server doesn't take responsibility for index settings.
                if (RunningMode.isNoInitMode()) {
                    exist = containsMapping;
                } else {
                    exist = containsMapping && structures.compareIndexSetting(tableName, createSetting(model));
                }
```

I think this should be better in perf.